pkg/services/serviceaccounts/api/token.go

```diff
@@ -160,6 +160,14 @@ func (api *ServiceAccountsAPI) CreateToken(c *models.ReqContext) response.Respon
 		}
 	}
 
+	if api.cfg.TokenExpirationDayLimit > 0 {
+		dayExpireLimit := time.Now().Add(time.Duration(api.cfg.TokenExpirationDayLimit) * time.Hour * 24).Truncate(24 * time.Hour)
```

Comment:
👋 Offering my thoughts as-is without expectation of any response or action:

`AddDate(0, 0, api.cfg.TokenExpirationDayLimit)` is a nice function that does the conversion from days to duration for you while handling daylight savings.

I would skip the truncation here. If I allow tokens for max 1 day and create a token at 23:59, it’ll have a one minute max lifetime afaict.

This reminds me that I really want the durations available to Grafana time ranges available from config as well 😁 It would be cool to be able to set this to `1y/fQ` and have it “valid for up to a year, rounded down to end of a fiscal quarter”. That is very much a sidenote.

Reply:
The part about the truncation gets it partially wrong, since you’re doing the truncation there as well. New theory: I might be allowed at 00:01 to have a 47h+ token and at 23:59 just the 24h token?

Reply:
Thank you for the suggestion with `AddDate`, I originally didn't use it because I still needed to add dates and truncate the time. The truncation is only used for validation on the backend, in theory, the `cmd.SecondsToLive` should be truncated already from the front end.

I didn't understand entirely your last comment. Can you elaborate on that?